### PR TITLE
impl(bigtable): add TracedRowReader

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #11508 

`RowReader` is basically a `StreamRange` with slight nuances.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11513)

### CODE REVIEW

File: google/cloud/bigtable/internal/traced_row_reader_test.cc [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  rows.reserve(row_keys.size());
+  for (auto& r : row_keys) {
+    rows.emplace_back(std::move(r), std::vector<bigtable::Cell>{});
+  }
+  return rows;
+}
+
+TEST(TracedStreamRange, Success) {
+  auto span_catcher = testing_util::InstallSpanCatcher();
+  bigtable::Row r{"r1", {}};
```
Comment (devbww) [inline google/cloud/bigtable/internal/traced_row_reader_test.cc:56]: Unused?

File: google/cloud/bigtable/internal/traced_row_reader_test.cc [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  for (auto const& v : traced) {
+    EXPECT_STATUS_OK(v);
+    EXPECT_FALSE(ThereIsAnActiveSpan());
+  }
+  EXPECT_FALSE(ThereIsAnActiveSpan());
+}
+
+TEST(TracedStreamRange, Cancel) {
+  auto span_catcher = testing_util::InstallSpanCatcher();
+  bigtable::Row r{"r1", {}};
```
Comment (devbww) [inline google/cloud/bigtable/internal/traced_row_reader_test.cc:136]: Unused?

File: google/cloud/bigtable/internal/traced_row_reader_test.cc (hunk 1 above)
Comment (dbolduc) [reply google/cloud/bigtable/internal/traced_row_reader_test.cc]: Fixed, thanks.

File: google/cloud/bigtable/internal/traced_row_reader_test.cc (hunk 2 above)
Comment (dbolduc) [reply google/cloud/bigtable/internal/traced_row_reader_test.cc]: Fixed, thanks.

**Review by devbww — APPROVED** [review]

**Pull request merged**